linkerd/app/inbound/src/http/tests.rs

```diff
@@ -630,7 +618,7 @@ fn hello_server(
 #[tracing::instrument]
 #[allow(deprecated)] // linkerd/linkerd2#8733
 fn grpc_status_server(
-    http: hyper::server::conn::Http,
+    server: hyper::server::conn::http2::Builder<TracingExecutor>,
```

Comment:
...and this, are the other core of this branch. we can tweak these types, without having to do any additional refactoring.

i'm punting the client end of this change into a separate branch, as mentioned in the description, because we currently share common helper functions for the http/1 and http/2 clients.